src/main/java/dgen/utils/serialization/PostgresSerializer.java

```diff
@@ -0,0 +1,221 @@
+            statement.append("\t");
+            statement.append(columnConfigToStatement(columnConfig, column));
+            statement.append(",\n");
+        }
+        statement.deleteCharAt(statement.length() - 2);
+
+        statement.append(");");
+        return statement.toString();
+    }
+
+    private String foreignKeyToStatement(String pkTable, String pkColumn, String fkTable, String fkColumn) {
+        StringBuilder statement = new StringBuilder();
+
+        statement.append("ALTER TABLE \"");
+        statement.append(databaseName + "\".\"" + fkTable + "\"\n");
+
+        statement.append("ADD FOREIGN KEY (\"");
+        statement.append(fkColumn);
+        statement.append("\") REFERENCES \"");
+        statement.append(databaseName + "\".\"" +pkTable + "\" (\"");
+        statement.append(pkColumn + "\");");
+
+        return statement.toString();
+    }
+
+    private String primaryKeyToStatement(String pkTable, String pkColumn) {
+        StringBuilder statement = new StringBuilder();
+
+        statement.append("ALTER TABLE \"");
+        statement.append(databaseName + "\".\"" + pkTable + "\"\n");
+
+        statement.append("ADD PRIMARY KEY (\"");
+        statement.append(pkColumn);
+        statement.append("\");");
+
+        return statement.toString();
+    }
+
+    private String insertTableValues(Table table) {
+        StringBuilder statement = new StringBuilder();
+
+        statement.append("INSERT INTO ");
+        statement.append("\"" + databaseName + "\".\"" + table.getAttributeName() + "\"");
+        statement.append(" VALUES\n");
+
+        List<Iterator<DataType>> columnIterators = new ArrayList<>();
+        List<Column> columns = table.getColumns();
+        for (Column c : columns) {
+            Iterator<DataType> it = c.getData().iterator();
+            columnIterators.add(it);
+        }
+
+        // record
+        boolean moreData = true;
+        while(moreData) {
+            List<String> recordValues = new ArrayList<>();
+            for (Iterator<DataType> it : columnIterators) {
+                if (! it.hasNext()) {
+                    moreData = false;
+                    continue;
+                }
+                DataType dt = it.next();
+
+                String value;
+                if (dt.nativeType() == NativeType.STRING) {
+                    value = "'" + dt.value() + "'";
+                } else {
+                    value = dt.value().toString();
+                }
+
+                recordValues.add(value);
+            }
+            String record = String.join(",", recordValues);
+
+            statement.append("(" + record + "),");
+            statement.append("\n");
+        }
+
+        statement.delete(statement.length() - 6, statement.length());
+        statement.append(";");
+
+        return statement.toString();
+    }
+
+    public static void main(String[] args) throws Exception {
```

Comment:
have you tested loading this into a postgres database? does the resulting schema look like what it should?

Reply:
Yes, it creates a new schema within the current database and loads the tabes, data, relationships, etc. properly

Reply:
Alright. Let me merge this PR and we will discuss the user interface later.